packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -988,6 +988,23 @@ const runGenerator = async (
                 mobify: {app: {extensions: [answers.project.name]}}
             })
 
+            // TODO: The generator is growing, we should refactor this to be more maintainable.
+            const processGeneratedExtension = () => {
+                // do a file content replacement for /src/setup-app.ts in the outputDir
+                // find all instances of "@salesforce/extension-base" and replace with answers.project.name
+                const setupAppPath = p.join(outputDir, 'src', 'setup-app.ts')
+                if (fs.existsSync(setupAppPath)) {
+                let setupAppContent = fs.readFileSync(setupAppPath, 'utf8')
+                    setupAppContent = setupAppContent.replace(
+                        /@salesforce\/extension-base/g,
+                        answers.project.name
+                    )
+                    fs.writeFileSync(setupAppPath, setupAppContent)
+                }
+            }
+
+            processGeneratedExtension()
```

Comment:
I'm not in love with this. But I'm also not going to die on a hill for beta. But would this be less ugly if we had the ID in an independant file that isn't the setup-app file? Probably. 

One idea is to just take the answers object and serialize that into the project folder. Then we can import the id from that file, whatever it ends up being called.

Reply:
I updated the implementation to use a `extension-meta.json` file, and the generator is also updated.